File: docs-examples/example-java/src/test/java/io/micronaut/mqtt/docs/parameters/ProductClient.java
package io.micronaut.mqtt.docs.parameters;

// tag::imports[]
import io.micronaut.mqtt.annotation.Topic;
import io.micronaut.mqtt.v5.annotation.MqttPublisher;
import io.micronaut.context.annotation.Requires;
// end::imports[]

@Requires(property = "spec.name", value = "BindingSpec")
// tag::clazz[]
@MqttPublisher
public interface ProductClient {

    @Topic("product") // <1>
    void send(byte[] data);

    void send(@Topic String binding, byte[] data); // <2>
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/mqtt/docs/parameters/ProductClient.py
from abc import ABC, abstractmethod
from typing import Annotated

# tag::imports[]
from micronaut.mqtt.annotation import Topic
from micronaut.mqtt.annotation.v5 import MqttPublisher
# end::imports[]

from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="BindingSpec")
# tag::clazz[]
@MqttPublisher
class ProductClient(ABC):

    @Topic("product")  # <1>
    @abstractmethod
    def send(self, data: bytes) -> None:
        ...

    @abstractmethod
    def send_to_topic(self, binding: Annotated[str, Topic], data: bytes) -> None:  # <2>
        ...
# end::clazz[]
